Clean up unnecessary else block and redundant variable assignment

myeunee opened the pull request:
This PR removes an unnecessary else block and eliminates a redundant variable assignment to improve code readability.

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
This PR removes an unnecessary `else` block and eliminates a redundant variable assignment to improve code readability and maintainability.


#### Which issue(s) this PR fixes:
None
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:
None

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs
NONE
```

myeunee committed:
Clean up unnecessary else block and redundant variable assignment

This PR removes an unnecessary else block and eliminates a redundant variable assignment to improve code readability.

myeunee commented:
Could you please run `/ok-to-test` and review this PR? Thank you! 😊

googs1025 commented:
/ok-to-test

googs1025 commented on cmd/kube-scheduler/app/server.go:
```diff
@@ -169,11 +168,12 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
-	if cz, err := configz.New("componentconfig"); err == nil {
-		cz.Set(cc.ComponentConfig)
-	} else {
+	cz, err := configz.New("componentconfig")
```
Hey, is there any logical change in the code that could make it easier to understand?

myeunee replied:
I removed the 'else' block because the 'if' statement already returns an error if the condition is met. 
By eliminating the 'else' block, it simplifies the code structure and avoids unnecessary nesting, which can improve readability.

googs1025 replied:
hey, but this doesn't seem to simplify the logic

thockin replied:
Yeah, I would lay this out differently than before or after:

```
if cz, err := configz.New("componentconfig"); err != nil {
    // return...
} else {
    // use cz
}
```

Our brains are wired to handle for `err != nil` - anything else is a hazard

myeunee replied:
Now I understand, thanks for the feedback.

myeunee replied:
Thank you for the feedback! I’ve refactored the code as you suggested to limit the scope of `cz` and `err` and to improve readability with the `if err != nil` pattern. I appreciate the guidance and the suggestions on Go best practices. Please let me know if there’s anything else to adjust.

myeunee commented on cmd/kube-scheduler/app/server.go:
```diff
@@ -133,13 +133,12 @@ for more information about scheduling and the kube-scheduler component.`,
-	fg := opts.ComponentGlobalsRegistry.FeatureGateFor(utilversion.DefaultKubeComponent)
 	// Activate logging as soon as possible, after that
 	// show flags with the final logging configuration.
-	if err := logsapi.ValidateAndApply(opts.Logs, fg); err != nil {
```
I initially changed it for consistency within the codebase by directly using 'opts.ComponentGlobalsRegistry.FeatureGateFor(utilversion.DefaultKubeComponent)' rather than introducing an additional variable 'fg'. 
However, I now understand that the original code was cleaner, so I will revert this part.

myeunee commented on cmd/kube-scheduler/app/server.go:
```diff
@@ -133,13 +133,12 @@ for more information about scheduling and the kube-scheduler component.`,
 		fmt.Fprintf(os.Stderr, "%v\n", err)
 		os.Exit(1)
-	}
+	}	
```
I understand. I will revert the change to keep the blank line for better readability.

myeunee committed:
Update server.go

Refactor: Removed unnecessary else block and redundant variable assignment

myeunee commented:
/test pull-kubernetes-verify

myeunee committed:
Apply gofmt formatting to server.go

myeunee committed:
Refactor error handling for configz initialization

Improved code readability and limited variable scope as per reviewer's suggestion.

myeunee commented:
/retest

myeunee commented:
/test pull-kubernetes-e2e-gce

thockin commented on cmd/kube-scheduler/app/server.go:
```diff
@@ -169,11 +169,10 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
+	if cz, err := configz.New("componentconfig"); err != nil {
 		return fmt.Errorf("unable to register configz: %s", err)
 	}
+	cz.Set(cc.ComponentConfig)
```
this still needs to be in an `else` - the lifetime of `cz` is the `if` clause.

myeunee replied:
Thank you for the feedback! I’ve updated the code to use an else clause, ensuring the scope of cz remains within the if block. Please let me know if there’s anything further to adjust.

myeunee committed:
Restrict cz variable scope within else clause

thockin reviewed (commented):
Thanks!

/lgtm
/approve
